src/main/java/ch/njol/skript/effects/EffWorldBorderExpand.java

```diff
@@ -0,0 +1,94 @@
+package ch.njol.skript.effects;
+
+import ch.njol.skript.Skript;
+import ch.njol.skript.doc.Description;
+import ch.njol.skript.doc.Examples;
+import ch.njol.skript.doc.Name;
+import ch.njol.skript.doc.Since;
+import ch.njol.skript.lang.Effect;
+import ch.njol.skript.lang.Expression;
+import ch.njol.skript.lang.SkriptParser.ParseResult;
+import ch.njol.skript.util.Timespan;
+import ch.njol.skript.util.Timespan.TimePeriod;
+import ch.njol.util.Kleenean;
+import org.bukkit.WorldBorder;
+import org.bukkit.event.Event;
+import org.jetbrains.annotations.Nullable;
+
+@Name("Expand/Shrink World Border")
+@Description({
+	"Expand or shrink the size of a world border.",
+	"Note: Using <code>by</code> adds/subtracts from the current size of the world border.",
+	"Using <code>to</code> sets to the specified size."
+})
+@Examples({
+	"expand world border of player by 100 in 5 seconds",
+	"shrink world border of world \"world\" to 100 in 10 seconds"
+})
+@Since("INSERT VERSION")
+public class EffWorldBorderExpand extends Effect {
+
+	static {
+		Skript.registerEffect(EffWorldBorderExpand.class,
+			"(expand|grow) [(diameter|:radius) of] %worldborders% (by|:to) %number% [over [[a ](time|period) of] %-timespan%]",
+			"(expand|grow) %worldborders%['s (diameter|:radius)] (by|:to) %number% [over [[a ](time|period) of] %-timespan%]",
+			"(contract|shrink) [(diameter|:radius) of] %worldborders% (by|:to) %number% [over [[a ](time|period) of] %-timespan%]",
+			"(contract|shrink) %worldborders%['s (diameter|:radius)] (by|:to) %number% [over [[a ](time|period) of] %-timespan%]"
+
+		);
+	}
+
+	private boolean shrink;
+	private boolean radius;
+	private boolean to;
+	private Expression<WorldBorder> worldBorders;
+	private Expression<Number> number;
+	private @Nullable Expression<Timespan> timespan;
+
+	@Override
+	@SuppressWarnings("unchecked")
+	public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelayed, ParseResult parseResult) {
+		worldBorders = (Expression<WorldBorder>) exprs[0];
+		number = (Expression<Number>) exprs[1];
+		timespan = (Expression<Timespan>) exprs[2];
+		shrink = matchedPattern > 1;
+		radius = parseResult.hasTag("radius");
+		to = parseResult.hasTag("to");
+		return true;
+	}
+
+	@Override
+	protected void execute(Event event) {
+		double input = number.getOptionalSingle(event).orElse(0).doubleValue();
+		if (radius)
+			input *= 2;
+		long speed = 0;
+		if (timespan != null) {
+			Timespan timespan = this.timespan.getSingle(event);
+			if (timespan != null)
+				speed = timespan.getAs(TimePeriod.SECOND);
+		}
+		WorldBorder[] worldBorders = this.worldBorders.getAll(event);
+		if (to) {
+			for (WorldBorder worldBorder : worldBorders)
+				worldBorder.setSize(Math.max(1, Math.min(input, 6.0E7)), speed);
```

Comment:
magic number, you should make this a private static final with a descriptive name

Reply:
You can also opt in for `Math2.fit(min, value, max)`